src/WorkflowStep.ts

```diff
@@ -0,0 +1,379 @@
+  const lastCallback = callbacks.pop();
+
+  if (lastCallback !== undefined) {
+    await processMiddleware(callbacks, args, context, client, logger, async () =>
+      lastCallback({ ...args, context, client, logger }),
+    );
+  }
+}
+
+export function isStepEvent(args: AnyMiddlewareArgs): args is AllWorkflowStepMiddlewareArgs {
+  const validTypes = new Set(['workflow_step_edit', 'workflow_step', 'workflow_step_execute']);
+  return validTypes.has(args.payload.type);
+}
+
+function selectToken(context: Context): string | undefined {
+  return context.botToken !== undefined ? context.botToken : context.userToken;
+}
+
+/**
+ * Factory for `configure()` utility
+ * @param args workflow_step_edit action
+ */
+function createStepConfigure(
+  args: AllWorkflowStepMiddlewareArgs<SlackActionMiddlewareArgs<WorkflowStepEdit>>,
+): StepConfigureFn {
+  const {
+    context,
+    client,
+    body: { callback_id, trigger_id },
+  } = args;
+  const token = selectToken(context);
+
+  return (config: Parameters<StepConfigureFn>[0]) => {
+    const { blocks, private_metadata, submit_disabled, external_id } = config;
+    return client.views.open({
+      token,
+      trigger_id,
+      view: {
+        callback_id,
+        blocks,
+        type: 'workflow_step',
+        // only include private_metadata, submit_disabled + external_id if passed by user
+        ...(private_metadata !== undefined && { private_metadata }),
+        ...(submit_disabled !== undefined && { submit_disabled }),
+        ...(external_id !== undefined && { external_id }),
+      },
+    });
+  };
+}
+
+/**
+ * Factory for `update()` utility
+ * @param args view_submission event
+ */
+function createStepUpdate(
+  args: AllWorkflowStepMiddlewareArgs<SlackViewMiddlewareArgs<ViewWorkflowStepSubmitAction>>,
+): StepUpdateFn {
+  const {
+    context,
+    client,
+    body: {
+      workflow_step: { workflow_step_edit_id },
+    },
+  } = args;
+  const token = selectToken(context);
+
+  return (config: Parameters<StepUpdateFn>[0]) => {
+    const { step_name, step_image_url, inputs = {}, outputs = [] } = config;
+    return client.workflows.updateStep({
+      token,
+      workflow_step_edit_id,
+      inputs,
+      outputs,
+      // only include step_name + step_image_url if passed by user
+      ...(step_name !== undefined && { step_name }),
+      ...(step_image_url !== undefined && { step_image_url }),
+    });
+  };
+}
+
+/**
+ * Factory for `complete()` utility
+ * @param args workflow_step_execute event
+ */
+function createStepComplete(
+  args: AllWorkflowStepMiddlewareArgs<SlackEventMiddlewareArgs<'workflow_step_execute'>>,
+): StepCompleteFn {
+  const {
+    context,
+    client,
+    payload: {
+      workflow_step: { workflow_step_execute_id },
+    },
+  } = args;
+  const token = selectToken(context);
+
+  return (config: Parameters<StepCompleteFn>[0]) => {
+    const { outputs = [] } = config;
```

Comment:
outputs should default to an empty object, as it's not an array, but a key/value mapping in the step execute portion.

Reply:
Going off of the other comments around config, since `outputs` is optional per the API docs, I altered this to be the following:

```
return (config: Parameters<StepCompleteFn>[0] = {}) => {
  return client.workflows.stepCompleted({
    token,
    workflow_step_execute_id,
    ...config,
  });
};
```

Let me know if this doesn't suit for some reason!